lib/cylc/network/client.py

```diff
@@ -231,9 +232,9 @@ def __new__(cls, suite, owner=None, host=None, port=None, timeout=None):
 
         # create connection
         return ZMQClient(
-            host, port, encrypt, decrypt, lambda: get_secret(suite),
+            host, port, encrypt, decrypt, partial(get_secret(suite)),
```

Comment:
I think is should be `partial(get_secret, suite)`. I think that's why Travis failed now.

```
+Traceback (most recent call last):
+  File "/home/travis/build/cylc/cylc/bin/cylc-scan", line 314, in <module>
+    main()
+  File "/home/travis/build/cylc/cylc/lib/cylc/terminal.py", line 65, in wrapper
+    function(*args, **kwargs)
+  File "/home/travis/build/cylc/cylc/bin/cylc-scan", line 194, in main
+    for reg, host, port, info in suites:
+  File "/home/travis/build/cylc/cylc/lib/cylc/network/scan.py", line 140, in scan_many
+    result for _, result in async_unordered_map(scan_one, args))
+  File "/home/travis/build/cylc/cylc/lib/cylc/network/scan.py", line 140, in <genexpr>
+    result for _, result in async_unordered_map(scan_one, args))
+  File "/home/travis/build/cylc/cylc/lib/cylc/network/scan.py", line 118, in async_unordered_map
+    yield (task.args, task.result())
+  File "/home/travis/build/cylc/cylc/lib/cylc/network/scan.py", line 160, in scan_one
+    client = SuiteRuntimeClient(reg, host=host, port=port, timeout=timeout)
+  File "/home/travis/build/cylc/cylc/lib/cylc/network/client.py", line 235, in __new__
+    host, port, encrypt, decrypt, partial(get_secret(suite)),
+TypeError: the first argument must be callable
+Task exception was never retrieved
```